grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/cache/Cache.java

```diff
@@ -153,9 +156,18 @@ public void add(Cache<Q, T> c2){
      * @return unified cached answers
      */
     public abstract T getAnswers(Q query);
+    //TODO: @Kasper Is this abstract method on `Cache` really needed?
+    //      It's only ever called through LazyQueryCache
+    @SuppressWarnings("unused")
     public abstract Pair<T, MultiUnifier> getAnswersWithUnifier(Q query);
     public abstract Stream<Answer> getAnswerStream(Q query);
+    //TODO: @Kasper Is this abstract method on `Cache` really needed?
+    //      It's only ever called through LazyQueryCache
+    @SuppressWarnings("unused")
     public abstract Pair<Stream<Answer>, MultiUnifier> getAnswerStreamWithUnifier(Q query);
+    //TODO: @Kasper Is this abstract method on `Cache` really needed?
```

Comment:
@kasper-piskorski

Reply:
@aelred had the same questions in his cleanup PR :D

Reply:
these are all implemented in `QueryCache` and `LazyQueryCache` so wanted to have a common parent

Reply:
If they not accessed through the abstraction then it's an unneeded inheritance. So I am going to kill it. Just shout if you totally disagree.